src/Agent.Listener/Configuration/CredentialProvider.cs

```diff
@@ -180,8 +159,49 @@ private Uri GetTenantAuthorityUrl(IHostContext context, string serverUrl)
                 }
             }
         }
-    }
 
+        /// <summary>
+        /// Gets an access token so that the application accesses the web api in the name of the user
+        /// who signs-in on a separate device
+        /// </summary>
+        /// <returns>An authentication result, or null if the user canceled sign-in, or did not sign-in on a separate device
+        /// after a timeout (15 mins)</returns>
+        private async Task<AuthenticationResult> GetTokenUsingDeviceCodeFlowAsync(IHostContext context, IPublicClientApplication app, IEnumerable<string> scopes)
+        {
+            AuthenticationResult result;
+            try
+            {
+                result = await app.AcquireTokenWithDeviceCode(scopes,
+                    deviceCodeCallback =>
+                    {
+                        // This will print the message on the console which tells the user where to go sign-in using 
+                        // a separate browser and the code to enter once they sign in.
+                        var term = context.GetService<ITerminal>();
+                        term.WriteLine($"Please finish AAD device code flow in browser ({deviceCodeCallback.VerificationUrl}), user code: {deviceCodeCallback.UserCode}"); return Task.FromResult(0);
+                    }).ExecuteAsync().ConfigureAwait(false);
+            }
+            catch (MsalServiceException ex)
+            {
+                string errorCode = ex.ErrorCode;
+                // AADSTS50059: No tenant-identifying information found in either the request or implied by any provided credentials.
+                // AADSTS90133: Device Code flow is not supported under /common or /consumers endpoint.
+                // AADSTS90002: Tenant <tenantId or domain you used in the authority> not found. This may happen if there are 
+                // no active subscriptions for the tenant. Check with your subscription administrator.
+                throw;
+            }
+            catch (OperationCanceledException)
+            {
+                result = null;
+            }
+            catch (MsalClientException ex)
+            {
+                string errorCode = ex.ErrorCode;
```

Comment:
`string errorCode` is unused both in there and above.

Reply:
Added trace instead